runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -152,14 +152,20 @@ inline void generateLoadJ9Class(TR::Node* node, TR::Register* j9class, TR::Regis
                    "Unexpected opCode for generateLoadJ9Class %s.", node->getOpCode().getName());
          break;
       }
+   if (node->getOpCode().isReadBar() || node->getOpCode().isWrtBar())
+      needsNULLCHK = true;
+
    auto use64BitClasses = TR::Compiler->target.is64Bit() && !TR::Compiler->om.generateCompressedObjectHeaders();
    auto instr = generateRegMemInstruction(LRegMem(use64BitClasses), node, j9class, generateX86MemoryReference(object, TR::Compiler->om.offsetOfObjectVftField(), cg), cg);
    if (needsNULLCHK)
       {
       cg->setImplicitExceptionPoint(instr);
       instr->setNeedsGCMap(0xFF00FFFF);
-      instr->setNode(cg->comp()->findNullChkInfo(node));
       }
+
+   if (opValue == TR::checkcastAndNULLCHK)
```

Comment:
We may also want to move this inside the earlier `if` statement. We don't need check the opcode when `needsNULLCHK` is false, since when `opValue == TR::checkcastAndNULLCHK` `needsNULLCHK` must be true.

Reply:
Changed according to comments.